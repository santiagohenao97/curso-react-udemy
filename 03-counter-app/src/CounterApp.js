import React, {useState} from 'react';
import PropTypes from 'prop-types'


const CounterApp = ({ value }) =>{

    const [ counter, setCounter ] = useState(value)
    
    const handleAdd = (e)=>{
        setCounter ( counter +1 )
        //setCounter( c => c+1) otra forma
    } 

    const handleSubstract = (e)=>{
        setCounter ( counter - 1 )
    } 

    const handleReset = (e)=>{
        setCounter ( value )
    } 

    return(
        <>
            <h1>Counter App</h1>
            <h2>{counter}</h2>

            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleReset }>Reset</button>
            <button onClick={ handleSubstract }>-1</button>
        </>
    )
}

CounterApp.propType ={
    saludo: PropTypes.number.isRequired
}

export default CounterApp