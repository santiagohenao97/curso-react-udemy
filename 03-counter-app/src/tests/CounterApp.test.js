import { shallow } from 'enzyme';
import React from 'react';
import CounterApp from '../CounterApp'

describe('Pruebas en <CounterApp />', () => {

    let wrapper = shallow(<CounterApp />);
    //Antes de todas las pruebas hace lo que está dentro del beforeEach
    beforeEach(()=>{
        wrapper = shallow(<CounterApp />);
    })

    test('debe mostrar <CounterApp/> correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe mostrar el valor por defecto de 100', () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value = {value}/>);
        const counterText = wrapper.find('h2').text().trim(); //trim elimina los espacios en blanco
        console.log(counterText);
        expect(counterText).toBe(`${value}`);
        
    })

    test('debe de incrementar con el botón +1', ()=>{
        
        wrapper.find('button').at(0).simulate('click'); //posición 1 es -1
        const counterText = wrapper.find('h2').text().trim()
        expect(counterText).toBe(`${parseInt(counterText) +1}`);

    })

    test('debe de decrementar con el botón -1', ()=>{
        
        wrapper.find('button').at(2).simulate('click'); //posición 1 es -1
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe(`${parseInt(counterText) -1}`);

    })

    test('debe reiniciar con el valor por defecto', () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value = {value}/>);
        wrapper.find('button').at(2).simulate('click'); //posición 1 es -1
        wrapper.find('button').at(1).simulate('click'); //posición del reset
        
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe(`${parseInt(value)}`);
    })
    
})
