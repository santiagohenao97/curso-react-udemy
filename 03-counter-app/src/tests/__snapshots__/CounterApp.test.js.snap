// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe mostrar <CounterApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2 />
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
